pkg/controller/appconfiguration/keyvalue.go

```diff
@@ -0,0 +1,195 @@
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.KeyValue{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.KeyValueGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube client.Client
+}
+
+func (c connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	content, _, err := azure.GetAuthInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, errors.Wrap(err, errConnectFailed)
+	}
+
+	cr, ok := mg.(*v1alpha1.KeyValue)
+	if !ok {
+		return &external{}, errors.New(errNotKeyValue)
+	}
+
+	clientArgs := keyvalues.NewClientAzureADArgs{
+		ClientID:         content["clientId"],
+		ClientSecret:     content["clientSecret"],
+		TenantID:         content["tenantId"],
+		AADEndpoint:      content["activeDirectoryEndpointUrl"],
+		ResourceEndpoint: cr.Spec.ForProvider.Endpoint,
+	}
+
+	cl, err := keyvalues.NewClientAzureAD(clientArgs)
+	if err != nil {
+		return nil, errors.Wrap(err, errCannotCreateNewClient)
+	}
+	return &external{kube: c.kube, client: cl}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client keyvalues.Client
+}
+
+func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.KeyValue)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotKeyValue)
+	}
+
+	key := cr.Spec.ForProvider.Key
+	label := cr.Spec.ForProvider.Label
+
+	kv, err := c.client.GetKeyValue(key, label)
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetFailed)
+	}
+
+	// Import/Sync
+	lateInit := false
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	appclient.LateInitialize(&cr.Spec.ForProvider, kv)
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := c.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+		lateInit = true
+	}
+
+	cr.Status.SetConditions(xpv1.Available())
+	appclient.GenerateObservation(&cr.Status.AtProvider, kv)
+
+	upToDate, err := appclient.IsUpToDate(cr.Spec.ForProvider, &kv)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errCheckUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        upToDate,
+		ResourceLateInitialized: lateInit,
+	}, nil
+}
+
+func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.KeyValue)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotKeyValue)
+	}
+
+	cr.Status.SetConditions(xpv1.Creating())
```

Comment:
crossplane-runtime can now [handle setting "creating" and "deleting" conditions](https://github.com/crossplane/crossplane-runtime/pull/242).